File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedGameStateChanged.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.constants.AmqRanked;
import tech.zolhungaj.amqapi.servercommands.CommandType;

@CommandType("ranked game state change")
public record RankedGameStateChanged(
        @Json(name = "serieId") AmqRanked.RankedSeries rankedSeries,
        @Json(name = "state") AmqRanked.RankedState state
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedGameStateChanged.kt
package tech.zolhungaj.amqapi.servercommands.globalstate

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.constants.RankedSeries
import tech.zolhungaj.amqapi.constants.RankedState
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.ActiveRankedGameModes

@JvmRecord
@CommandType("ranked game state change")
data class RankedGameStateChanged(
    @Json(name = "serieId") val rankedSeries: RankedSeries,
    @Json(name = "state") val state: RankedState,
    @Json(name = "games") val activeGames: ActiveRankedGameModes,
)
